crates/exex/exex/src/backfill.rs

```diff
@@ -185,41 +185,70 @@ where
         let chain = Chain::new(blocks, executor.finalize(), None);
         Ok(chain)
     }
+
+    /// Returns an iterator that executes each block in the range separately and yields the
+    /// resulting [`Block`] and [`ExecutionOutcome`].
+    pub fn into_single_blocks(
+        &self,
+    ) -> impl Iterator<Item = Result<(Block, ExecutionOutcome), BlockExecutionError>> + '_ {
+        self.range.clone().map(|block_number| {
+            let td = self
+                .provider
+                .header_td_by_number(block_number)?
+                .ok_or_else(|| ProviderError::HeaderNotFound(block_number.into()))?;
+
+            // Fetch the block with senders for execution.
+            let block_with_senders = self
+                .provider
+                .block_with_senders(block_number.into(), TransactionVariant::WithHash)?
+                .ok_or_else(|| ProviderError::HeaderNotFound(block_number.into()))?;
+
+            // Configure the executor to use the previous block's state.
+            let mut executor = self.executor.batch_executor(
+                StateProviderDatabase::new(
+                    self.provider.history_by_block_number(block_number.saturating_sub(1))?,
+                ),
+                self.prune_modes.clone(),
+            );
+
+            trace!(target: "exex::backfill", number = block_number, txs = block_with_senders.block.body.len(), "Executing block");
+
+            executor
+                .execute_and_verify_one((&block_with_senders, td).into())?;
+
+            Ok((block_with_senders.block, executor.finalize()))
```

Comment:
let's return block with senders, not just block
```suggestion
            Ok((block_with_senders, executor.finalize()))
```